src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockDepths/boss_magmus.cpp

```diff
@@ -110,7 +117,61 @@ class boss_magmus : public CreatureScript
         }
 };
 
+class npc_ironhand_guardian : public CreatureScript
+{
+    public:
+
+        npc_ironhand_guardian() : CreatureScript("npc_ironhand_guardian") { }
+
+        struct npc_ironhand_guardianAI : public ScriptedAI
+        {
+
+            npc_ironhand_guardianAI(Creature* creature) : ScriptedAI(creature)
+            {
+                _instance = me->GetInstanceScript();
```

Comment:
missing initialize _lastState in here.

Reply:
Not realy needed, but you can use octor list to initialize it.
like:
`
npc_ironhand_guardianAI(Creature* creature) : ScriptedAI(creature), _instance(creature->GetInstanceScript(),_lastState(0) { }
`

Reply:
@martintopholm